test/powershell/SDK/PSDebugging.Tests.ps1

```diff
@@ -184,5 +185,94 @@ Describe "PowerShell Command Debugging" -tags "CI" {
 
 }
 
+# Scripting\Debugging\RunspaceDebuggingTests.cs
+Describe "Runspace Debugging API tests" -tag CI {
+    Context "PSStandaloneMonitorRunspaceInfo tests" {
+        BeforeAll {
+            $runspace = [runspacefactory]::CreateRunspace()
+            $runspaceType = [PSMonitorRunspaceType]::WorkflowInlineScript
+            $monitorInfo = [PSStandaloneMonitorRunspaceInfo]::new($runspace)
+            $instanceId = $runspace.InstanceId
+            $parentDebuggerId = [guid]::newguid()
+            $ps = [powershell]::Create()
+            $embeddedRunspaceInfo = [PSEmbeddedMonitorRunspaceInfo]::New($runspace,$runspaceType,$ps, $parentDebuggerId)
+        }
+        AfterAll {
+            $runspace.Dispose()
+            $ps.dispose()
+        }
+        
+        It "PSStandaloneMonitorRunspaceInfo should throw when called with a null argument to the constructor" {
+            try {
+                [PSStandaloneMonitorRunspaceInfo]::new($null)
+                throw "Execution should have thrown"
+            }
+            Catch {
+                $_.FullyQualifiedErrorId | should be PSArgumentNullException
+            }
+        }
+
+        it "PSStandaloneMonitorRunspaceInfo properties should have proper values" {
+            $monitorInfo.Runspace.InstanceId | Should be $InstanceId
+            $monitorInfo.RunspaceType | Should be "StandAlone"
+            $monitorInfo.NestedDebugger | Should BeNullOrEmpty
+        }
+
+        It "Embedded runspace properties should have proper values" {
+            $embeddedRunspaceInfo.Runspace.InstanceId | should be $InstanceId
+            $embeddedRunspaceInfo.ParentDebuggerId | should be $parentDebuggerId
+            $embeddedRunspaceInfo.Command.InstanceId | should be $ps.InstanceId
+            $embeddedRunspaceInfo.NestedDebugger | Should BeNullOrEmpty
+        }
+    }
+    Context "Test Monitor RunspaceInfo API tests" {
+        BeforeAll {
+            $runspace = [runspacefactory]::CreateRunspace()
+            $runspace.Open()
+            $associationId = [guid]::newguid()
+            $runspaceInfo = [PSStandaloneMonitorRunspaceInfo]::new($runspace)
+        }
```

Comment:
quite right

---
In reply to: [78606708](https://github.com/PowerShell/PowerShell/pull/2239#discussion_r78606708) [](ancestors = 78606708)